opm/autodiff/BlackoilPropsAdFromDeck.cpp

```diff
@@ -141,18 +146,41 @@ namespace Opm
                 if (numSamples > 0) {
                     auto splinePvt = std::shared_ptr<PvtDeadSpline>(new PvtDeadSpline);
                     splinePvt->initFromOil(pvdoTables, numSamples);
+
+                    if (!eclState->getOilvisctTables().empty()) {
+                        splinePvt->setOilvisctTables(eclState->getOilvisctTables(),
+                                                     deck->getKeyword("VISCREF"));
+                    }
+
                     props_[phase_usage_.phase_pos[Liquid]] = splinePvt;
                 } else {
                     auto deadPvt = std::shared_ptr<PvtDead>(new PvtDead);
                     deadPvt->initFromOil(pvdoTables);
+
+                    if (!eclState->getOilvisctTables().empty()) {
+                        deadPvt->setOilvisctTables(eclState->getOilvisctTables(),
+                                                   deck->getKeyword("VISCREF"));
+                    }
+
                     props_[phase_usage_.phase_pos[Liquid]] = deadPvt;
                 }
             } else if (!pvtoTables.empty()) {
-                props_[phase_usage_.phase_pos[Liquid]].reset(new PvtLiveOil(pvtoTables));
```

Comment:
that is because some methods specific to `PvtLiveOil` need to be called and the `props_` vector stores `BlackOilPvtInterface` pointers. At your option I can convert the raw 'pvt' pointer to a 'std::shared_ptr' and rename it to `pvto`, but I don't think this improves code quality much...